plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -382,7 +382,8 @@ public static Location buildSplitPath(Location tableLocation, AddFileEntry addAc
         Optional<String> scheme = tableLocation.scheme();
         if (scheme.isPresent() && (scheme.get().equals("abfs") || scheme.get().equals("abfss"))) {
             // Replace '+' with '%2B' beforehand. Otherwise, the character becomes a space ' ' by URL decode.
-            path = URLDecoder.decode(path.replace("+", "%2B"), UTF_8);
+            // Replace '%2F' with '%252F' in order to maintain also after URL decoding the original '%2F'
+            path = URLDecoder.decode(path.replace("+", "%2B").replace("%2F", "%252F"), UTF_8);
```

Comment:
why isn't this solved in the filesystem layer?

Reply:
It is resolved now - i've taken the contribution from https://github.com/trinodb/trino/pull/17264